validation/kill_no_effect.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"time"
+
+	"github.com/mndrix/tap-go"
+	rspecs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-tools/specerror"
+	"github.com/opencontainers/runtime-tools/validation/util"
+	uuid "github.com/satori/go.uuid"
+)
+
+func main() {
+	t := tap.New()
+	t.Header(0)
+	bundleDir, err := util.PrepareBundle()
+	if err != nil {
+		util.Fatal(err)
+	}
+	defer os.RemoveAll(bundleDir)
+
+	targetErr := specerror.NewError(specerror.KillNonCreateRunHaveNoEffect, fmt.Errorf("attempting to send a signal to a container that is neither `created` nor `running` MUST have no effect on the container"), rspecs.Version)
+	containerID := uuid.NewV4().String()
+	g := util.GetDefaultGenerator()
+	g.SetProcessArgs([]string{"true"})
+
+	config := util.LifecycleConfig{
+		Config:    g,
+		BundleDir: bundleDir,
+		Actions:   util.LifecycleActionCreate | util.LifecycleActionStart | util.LifecycleActionDelete,
+		PreCreate: func(r *util.Runtime) error {
+			r.SetID(containerID)
+			return nil
+		},
+		PreDelete: func(r *util.Runtime) error {
+			err := util.WaitingForStatus(*r, util.LifecycleStatusStopped, time.Second*5, time.Second*1)
+			if err != nil {
+				return err
+			}
+			currentState, err := r.State()
+			if err != nil {
+				return err
+			}
+			r.Kill("KILL")
```

Comment:
You probably want to ensure that this call returns an error, based on [this][1] and [this][2].

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.1/runtime.md#L125
[2]: https://github.com/opencontainers/runtime-tools/blame/v0.5.0/docs/command-line-interface.md#L211

Reply:
```
* *Exit code:* Zero if the signal was successfully sent to the container process and non-zero on errors.
```
I found there is no general exit code,  there is golang syscall pkg, but just works in posix system.
So do we still need this 'Exit code' session in command-line-interface?
```
if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
                log.Printf("Exit Status: %d", status.ExitStatus())
}
```

Reply:
The return error is check in `kill.go`. There is a bug in the previous kill.go, it displays the wrong spec error code. The new commit 0ddb5cd solves that.